wdl-analysis/src/analyzer.rs

```diff
@@ -393,52 +423,79 @@ impl Analyzer {
         }
     }
 
-    /// Finds documents to analyze.
+    /// Adds documents to the analyzer.
     ///
     /// If a specified path is a directory, it is recursively searched for WDL
     /// documents.
     ///
-    /// Returns the URIs of the discovered documents.
-    pub async fn find_documents(paths: Vec<PathBuf>) -> Vec<Arc<Url>> {
-        RayonHandle::spawn(move || {
+    /// Returns an error if there was a problem discovering documents for the
+    /// specified paths.
+    pub async fn add_documents(&self, paths: Vec<PathBuf>) -> Result<()> {
+        // Start by searching for documents
+        let documents = RayonHandle::spawn(move || {
             let mut documents = Vec::new();
             for path in paths {
                 if path.is_file() {
-                    if let Some(uri) = path_to_uri(&path) {
-                        documents.push(uri.into());
-                    }
-
+                    documents.push(path_to_uri(&path).with_context(|| {
+                        format!(
+                            "failed to convert path `{path}` to a URI",
+                            path = path.display()
+                        )
+                    })?);
                     continue;
                 }
 
+                if !path.is_dir() {
```

Comment:
What about `PathBuf::is_symlink()`?

Reply:
Both `is_file` and `is_dir` traverse symlinks so the only case in which both return false is an error to retrieve the filesystem metadata.

I'll also rewrite this to be slightly more efficient as only a single metadata call is needed.